### ISSUE TITLE
 the message saying "A matching Triton is not available," is slowing down the learning. 

### ISSUE THREAD

**Issue by kimura-daigo-boi** [kohya-ss/sd-scripts#829]
Hi, I am currently trying to speed up my learning process, but T keep getting the triton error which is slowing the whole process down. can anyone help me with this? 
i began the speedup with
・updating the cloned repo to the newest version on my pc
・updated xformers to 0.0.20,
・updated torch to torch-2.0.1+cu118 torchaudio-2.0.2+cu118 torchvision-0.15.2+cu118
although each learning step became much faster and the code seems to be working fine, now when I execute my sd-scripts, I keep getting the message 
`A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'`
multiple times whenever the code iterated to a new epoch. each time this shows it takes about 5~10 secs and it adds up to be a lot in the long run. This (obviously) becomes a pretty annoying problem when I have more than few epochs. 
the following is an example for what I get. 

running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 60
  num reg images / 正則化画像の数: 305
  num batches per epoch / 1epochのバッチ数: 60
  num epochs / epoch数: 37
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 2190
steps:   0%|                                                              | 0/2190 [00:00<?, ?it/s]
epoch 1/37
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
steps:   3%|█                                       | 60/2190 [01:00<35:47,  1.01s/it, loss=0.0967]
epoch 2/37
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
steps:   5%|██▏                                    | 120/2190 [02:00<34:34,  1.00s/it, loss=0.0958]
generating sample images at step / サンプル画像生成 ステップ: 120
prompt: vrdhd, 1 girl, walking in streets, upper body,  masterpiece, high quality,
negative_prompt: (low quality, worst quality), bad anatomy, bad composition, poor, monochrome, blandheight: 512
width: 512
sample_steps: 30
scale: 7.5
100%|██████████████████████████████████████████████████████████████| 30/30 [00:05<00:00,  5.65it/s]
100%|██████████████████████████████████████████████████████████████| 30/30 [00:05<00:00,  5.64it/s]
epoch 3/37

btw i use a 3060 12g for this. 
i tried installing the higgingface-resolved triton https://huggingface.co/r4ziel/xformers_pre_built/resolve/main/triton-2.0.0-cp310-cp310-win_amd64.whl  but that only deleted the message and did not do anything to speed up the process.

**Comment by imD-5** [comment]
I'm running it on windows. 
I am aware of the triton not being available for windows, but that wasn't the main problem here. the problem was that whatever that's causing this message top repeat is slowing down the learning significantly.

**Comment by imD-5** [comment]
oh i forgot to mention but this is happenening when im training loras

**Comment by FurkanGozukara** [comment]
> --persistent_data_loader_workers

i didn't know this

we should always use this argument? any side or negative effects?

**Comment by kimura-daigo-boi** [comment]
this solved the problem. thanks!

**Comment by FurkanGozukara** [comment]
> Is there a simple fix to just change the file and not have `Triton` called? If we can't use it, why keep it?

it works on linux
that is why linux training faster

**Comment by FurkanGozukara** [comment]
> > > Is there a simple fix to just change the file and not have `Triton` called? If we can't use it, why keep it?
> > 
> > 
> > it works on linux that is why linux training faster
> 
> Thank you. Understood. People just seem to have a problem with the "error" it shows. Myself included. It took me a while to find information on why it wouldn't work on my system (windows).
> 
> I'm asking if the users on windows can make this change to the file that calls for triton.

it is not an error that breaks anything
